### DISCUSSION TITLE
Authentication needs for Apollo Explorer

### DISCUSSION THREAD

**Discussion by ndintenfass** [apollographql/apollo-studio-community#45] [General]
We have an ongoing project to expand the options for how you and your users can authenticate in Apollo Explorer. Tell us how your graph works in terms of authentication and what blocks your team from using Explorer -- eg: do you need an oAuth integration? Do you need a way to secure your tokens away from the client?

**Comment by dmiyamasu** [comment]
@ndintenfass the biggest hurdle for us is the licensing model of Admin vs Consumer, with each Admin increasing the cost of using Apollo Studio.  The Consumer user has so little access and features in the Explorer and Studio that our developers are not using the Consumer accounts.

**Reply by ndintenfass** [reply]
Thanks for this feedback. We are looking into ways to smooth out the concern.

**Comment by setchy** [comment]
For us it would be having OAuth2 integration, appropriate JWT token caching and a method whereby the client/secrets are secured and defined per variant/contract.

Also, having something similar to an identity/authentication collection (set of pre-defined client/secret combos with different read-only permissions/scopes) that allow Graph Consumers to easily pick between a few different personas when using Explorer or Sandbox.
